### ISSUE TITLE
`no-duplicate-string` rule false positive on "isObjectPropertyKey" of Proxy

### ISSUE THREAD

**Issue by JeremyJonas** [SonarSource/eslint-plugin-sonarjs#361]
The `no-duplicate-string` rule triggers **FALSE POSITIVE** for object property key of Proxy.

https://github.com/SonarSource/eslint-plugin-sonarjs/blob/b211bf18a9afe22e2e36c48a25b983d4db5312a9/src/rules/no-duplicate-string.ts#L129-L131

## Repro Steps
1. Create any [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) object
2. Use string literal key to access properties of the Proxy object multiple times in file (3 for default threshold)

```
const myProxy = new Proxy({},{});
myProxy['Foo-Bar'];
myProxy['Foo-Bar'];
myProxy['Foo-Bar'];
myProxy['Foo-Bar'];
myProxy['Foo-Bar'];
```

**Comment by electrovir** [comment]
This is a problem with _any_ object that happens (for whatever reason) to have string properties that require access via bracket notation `[]` (as shown above) vs dot notation `.`:

```typescript
const myObject = {
    'my-property': 4,
    myProperty: 5,
};

myObject['my-property'];
myObject['my-property'];
myObject['my-property'];
// fails no-duplicate-string
myObject['my-property'];

myObject.myProperty;
myObject.myProperty;
myObject.myProperty;
// does not fail no-duplicate-string (obviously)
myObject.myProperty;
```

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-24](https://sonarsource.atlassian.net/browse/ESLINTJS-24)
